M2/Macaulay2/tests/normal/augmented-assignment.m2

```diff
@@ -74,6 +74,15 @@ assert Equation(x#0, 28)
 Foo @ Foo := (x, y) -> Foo(4*x#0 - 5*y#0)
 x @= Foo 7
 assert Equation(x#0, 77)
+Foo · Foo := (x,y) -> Foo(6*x#0 * 7*y#0)
+x ·= Foo 8
```

Comment:
This looks awkward, and I'm not convinced that it is useful enough to justify supporting it.

Reply:
My original thought was that every flexible binary operator (as long as it didn't introduce ambiguity when appending an extra `=`, like `==` -> `===`) should support augmented assignment out of the box.

While it's certainly true it would probably be a bad idea to use `·=` for an augmented dot product (First `v` is a vector, but after `v ·= w`, it's an element of the ring...), we might as well make it available for some other context where it might make sense.